control-plane/pkg/reconciler/trigger/triggerv2.go

```diff
@@ -0,0 +1,180 @@
+	EventingClient      eventingclientset.Interface
+	Env                 *config.Env
+	ConsumerGroupLister internalslst.ConsumerGroupLister
+	InternalsClient     internalsclient.Interface
+}
+
+func (r *ReconcilerV2) ReconcileKind(ctx context.Context, trigger *eventing.Trigger) reconciler.Event {
+	logger := kafkalogging.CreateReconcileMethodLogger(ctx, trigger)
+
+	broker, err := r.BrokerLister.Brokers(trigger.Namespace).Get(trigger.Spec.Broker)
+	if err != nil && !apierrors.IsNotFound(err) {
+		trigger.Status.MarkBrokerFailed("Failed to get broker", "%v", err)
+		return fmt.Errorf("failed to get broker: %w", err)
+	}
+
+	if apierrors.IsNotFound(err) {
+		// Actually check if the broker doesn't exist.
+		// Note: do not introduce another `broker` variable with `:`
+		broker, err = r.EventingClient.EventingV1().Brokers(trigger.Namespace).Get(ctx, trigger.Spec.Broker, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get broker: %w", err)
+		}
+	}
+
+	// Ignore Triggers that are associated with a Broker we don't own.
+	if isKnativeKafkaBroker, brokerClass := isKnativeKafkaBroker(broker); !isKnativeKafkaBroker {
+		logger.Debug("Ignoring Trigger", zap.String(eventing.BrokerClassAnnotationKey, brokerClass))
+		return nil
+	}
+
+	trigger.Status.PropagateBrokerCondition(broker.Status.GetTopLevelCondition())
+
+	if !broker.IsReady() {
+		// Trigger will get re-queued once this broker is ready.
+		return nil
+	}
+
+	consgroup, err := r.reconcileConsumerGroup(ctx, trigger)
+	if err != nil {
+		trigger.Status.MarkDependencyFailed("failed to reconcile consumer group", err.Error())
+		return err
+	}
+	trigger.Status.MarkDependencySucceeded()
+
+	trigger.Status.SubscriberURI = consgroup.Status.SubscriberURI
+	trigger.Status.MarkSubscriberResolvedSucceeded()
+
+	trigger.Status.DeadLetterSinkURI = consgroup.Status.DeadLetterSinkURI
+	trigger.Status.MarkDeadLetterSinkResolvedSucceeded()
+
+	return nil
+}
+
+func (r ReconcilerV2) reconcileConsumerGroup(ctx context.Context, trigger *eventing.Trigger) (*internalscg.ConsumerGroup, error) {
+
+	var deliveryOrdering = internals.Unordered
+	var err error
+	deliveryOrderingAnnotationValue, ok := trigger.Annotations[deliveryOrderAnnotation]
+	if ok {
+		deliveryOrdering, err = deliveryOrderingFromString(deliveryOrderingAnnotationValue)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	//todo additionally handle trigger.Spec.Filter
+	newcg := &internalscg.ConsumerGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      string(trigger.UID),
+			Namespace: trigger.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*kmeta.NewControllerRef(trigger),
+			},
+		},
+		Spec: internalscg.ConsumerGroupSpec{
+			Template: internalscg.ConsumerTemplateSpec{
+				Spec: internalscg.ConsumerSpec{
+					Topics:  []string{}, //todo from contract resource
+					Configs: internalscg.ConsumerConfigs{Configs: map[string]string{"group.id": string(trigger.UID)}},
```

Comment:
We should also set `bootstrap.servers` that we, ideally, get from the broker status.

Reply:
Same as above